### PR TITLE
Add Watchdog for iMXRT

### PR DESCRIPTION
**Pull request by HarrisonChoi750405**
1. add watchdog api
2. add hw reset function

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [40e68274c7551f2cc220669c0b12e3fd38fc1941] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [40e68274c7551f2cc220669c0b12e3fd38fc1941] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_wdog.c:95: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:98: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:98: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_wdog.c:99: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:100: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:100: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:100: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_wdog.c:101: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:101: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:102: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:102: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:103: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:103: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:104: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:106: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:108: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:133: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:136: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:138: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:139: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:140: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:141: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:142: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:143: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:144: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:145: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:146: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:147: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:169: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:171: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:173: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:174: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:174: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:175: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:175: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:176: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:176: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:180: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:183: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:184: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:185: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:186: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:186: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_wdog.c:187: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:188: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:188: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:189: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:202: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:202: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_wdog.c:203: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:204: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:204: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:205: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:224: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:226: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:227: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:230: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:232: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:251: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:251: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_wdog.c:252: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:253: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_wdog.c:253: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:254: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:267: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.c:268: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:216: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:243: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:259: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:275: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:323: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:338: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:351: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [0d61e036e7b5f715c7e8bf3befe5987283d01ad2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0d61e036e7b5f715c7e8bf3befe5987283d01ad2] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_wdog.h:216: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:243: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:259: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:275: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:323: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:338: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_wdog.h:351: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [f083c47de495fc778ffa342df74231222b49e97b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f083c47de495fc778ffa342df74231222b49e97b] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_wdog.h [hunk 1]
```diff
@@ -68,6 +68,300 @@
...
+ * This function writes a value into WCR registers.
+ * The time-out value can be written at any point of time but it is loaded to the counter at the time
+ * when WDOG is enabled or after the service routine has been performed.
+ *
+ * @param base WDOG peripheral base address
+ * @param timeoutCount WDOG timeout value; count of WDOG clock tick.
+ */
+static inline void imxrt_wdog_settimeoutvalue(WDOG_Type *base, uint16_t timeoutCount)
+{
+base->WCR = (base->WCR & ~WDOG_WCR_WT_MASK) | WDOG_WCR_WT(timeoutCount);
```
Comment (amandeepc) [inline os/arch/arm/src/imxrt/imxrt_wdog.h]: Please add space/tab here

**Comment by seinfra** [comment]
Target : [43df68880bbc788e2bebcd6ff577aa03d3ceccd8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [43df68880bbc788e2bebcd6ff577aa03d3ceccd8] - Code Rule Check (C++) OK.

**Pull request merged**